### PR TITLE
Merged with master and fixed compile time errors

### PR DESCRIPTION
**Pull request by shivamvats**
I have removed the move constructor. hash_set seems to have some issues with it

### CODE REVIEW

File: symengine/dict.h [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 #include <gmpxx.h>
 #if defined(HAVE_SYMENGINE_PIRANHA)
-#include <piranha/piranha.hpp>
+#include <piranha.hpp>
```
Comment (sumith1896) [inline symengine/dict.h]: @shivamvats I think this is left-over

Comment (shivamvats) [inline symengine/dict.h]: Ah yes!
I didn't get what you suggested on gitter (to fix this). Could you elaborate?

File: symengine/polys.h [hunk 2]
```diff
@@ -60,9 +69,9 @@ RCP<const Polynomial> sub_poly(const Polynomial &a, const Polynomial &b);
 //! Multiplying two Polynomial a and b
 RCP<const Polynomial> mul_poly(RCP<const Polynomial> a, RCP<const Polynomial> b);
 
-inline RCP<const Polynomial> polynomial(const vec_symbol &vars, hash_set &&polys_set)
+inline RCP<const Polynomial> polynomial(const vec_symbol &vars, hash_set &polys_set)
 {
-    return make_rcp<const Polynomial>(vars, std::move(polys_set));
+    return make_rcp<const Polynomial>(vars, polys_set);
```
Comment (sumith1896) [inline symengine/polys.h:74]: @shivamvats Not necessary I think, it works fine

File: symengine/dict.h (hunk 1 above)
Comment (sumith1896) [inline symengine/dict.h]: Yes, will do in gitter

**Comment by sumith1896** [comment]
Why are all commits appearing, I'm really worried about them.
I'm afraid, if I merge this, all the commits will appear in the main PR too, which is not the case with regular merge with master

**Comment by sumith1896** [comment]
Fixed via https://github.com/Sumith1896/csympy/commit/1b47817d232c0dc06638187756b97b8ffc89c872 

Builds and tests pass

Thanks for the help @shivamvats , sorry to be closing this

**Comment by sumith1896** [comment]
Let me know what changes you need there

File: symengine/polys.h (hunk 2 above)
Comment (shivamvats) [inline symengine/polys.h:74]: You have not used move constructor in your tests. Make the change there too, if you want to use it

**Comment by shivamvats** [comment]
No problem.
I think the extra commits appear because your branch is not up-to date with master.

**Pull request closed without merging**